### PR TITLE
registration button on profile tab, widget for avatar

### PR DESCRIPTION
**Pull request by rubenhorn**
~@Xazin The profile does not get refreshed after registration, but I couldn't figure out what causes this.~

### CODE REVIEW

File: lib/presentation/profile/profile_screen.dart [hunk 1]
```diff
@@ -21,9 +24,10 @@ class _UserProfilePageState extends State<UserProfilePage> {
   @override
   Widget build(BuildContext context) {
     final bioController = TextEditingController();
+    final profileBloc = getIt<ProfileBloc>();
```
Comment (Xazin) [inline lib/presentation/profile/profile_screen.dart:27]: If we don't pass it to another screen or widget, then we shouldn't initialize it in the build method and should just use it.

File: lib/presentation/profile/profile_screen.dart [hunk 2]
```diff
@@ -287,6 +283,17 @@ class _UserProfilePageState extends State<UserProfilePage> {
...
                             ],
+                          ] else ...[
+                            // TODO only for MVP (remove later)
+                            const SizedBox(height: 40),
+                            PillButton(
+                              text: 'Create account or sign in',
+                              onTap: () async {
+                                await context.router.push(const AuthRoute());
+                                // Refresh profile
+                                profileBloc.add(GetUserProfile());
```
Comment (Xazin) [inline lib/presentation/profile/profile_screen.dart:294]:
```suggestion
                                BlocProvider.of<ProfileBloc>(context).add(GetUserProfile());
```
or 
```suggestion
                                context.read<ProfileBloc>().add(GetUserProfile());
```

File: lib/presentation/profile/widget/profile_picture.dart [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    return CircleAvatar(
+      maxRadius: _maxRadius,
+      foregroundImage: imageProvider,
+      backgroundImage: const AssetImage(
+        'assets/images/logo.png',
+      ),
+    );
+  }
```
Comment (Xazin) [inline lib/presentation/profile/widget/profile_picture.dart:39]:
```suggestion
  Widget build(BuildContext context) {
    return CircleAvatar(
      maxRadius: _maxRadius,
      foregroundImage: _image != null ? FileImage(_image!) as ImageProvider<Object> :   NetworkImage(
        'https://static-dev.collaction.org/profile-pictures/$_userId.png'),
      onForegroundImageError: (_, __) => const AssetImage(
        'assets/images/logo.png',
      ),
    );
  }
```
I think this looks better.

File: lib/presentation/profile/profile_screen.dart (hunk 1 above)
Comment (rubenhorn) [reply lib/presentation/profile/profile_screen.dart:27]: This change actually seemed to fix the issue of the profile not being reloaded after login. (`profileBloc` is used after pushing the auth route)

File: lib/presentation/profile/widget/profile_picture.dart (hunk 3 above)
Comment (rubenhorn) [reply lib/presentation/profile/widget/profile_picture.dart:39]: I would still prefer handling `_userId == null` explicitly than hitting our backend with invalid requests.

Comment (Xazin) [reply lib/presentation/profile/widget/profile_picture.dart:39]:
```suggestion
  Widget build(BuildContext context) {
    return CircleAvatar(
      maxRadius: _maxRadius,
      foregroundImage: 
        _image != null 
        ? FileImage(_image!) as ImageProvider<Object> 
        :  _userId != null 
          ?  NetworkImage('https://static-dev.collaction.org/profile-pictures/$_userId.png')
          : const AssetImage('assets/images/logo.png'),
      onForegroundImageError: (_, __) => const AssetImage('assets/images/logo.png'),
    );
  }
```

File: lib/presentation/profile/profile_screen.dart (hunk 2 above)
Comment (rubenhorn) [reply lib/presentation/profile/profile_screen.dart:294]: Build context should not be used across `async`!

**Review by Xazin — APPROVED** [review]

**Pull request merged**